.github/actions/setup-and-login/action.yml

```diff
@@ -0,0 +1,67 @@
+name: Setup Node, Env and login to GHCR
+
+inputs:
+  install-packages:
+    required: false
+    default: 'false'
+  build-and-run-server:
+    required: false
+    default: 'false'
+  ghcr-username:
+    required: false
+    default: ''
+  ghcr-password:
+    required: false
+    default: ''
+
+outputs:
+  CI_SHA:
+    description: "Short SHA of current commit"
+    value: ${{ steps.ci-env.outputs.CI_SHA }}
+  CI_BRANCH:
+    description: "Current branch"
+    value: ${{ steps.ci-env.outputs.CI_BRANCH }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Login to GHCR (GitHub Packages)
+      if: inputs.ghcr-username != '' && inputs.ghcr-password != ''
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ inputs.ghcr-username }}
+        password: ${{ inputs.ghcr-password }}
+    - id: ci-env
+      name: Setup Environment
+      shell: bash
+      run: |
+        if [ "${{github.event_name}}" = "pull_request" ];
+        then
+          long_sha=${{ github.event.pull_request.head.sha }}
+          echo "CI_BRANCH=${{ github.head_ref }}" >> $GITHUB_OUTPUT
+        else
+          long_sha=${GITHUB_SHA}
+          echo "CI_BRANCH=${{ github.ref_name }}" >> $GITHUB_OUTPUT
+        fi
+        echo "CI_SHA=${long_sha:0:7}" >> $GITHUB_OUTPUT
+
+    - uses: actions/setup-node@v4
+      with:
+        node-version: 20
+
+    - name: Install packages
+      # we install in the build script anyway so skip if build-and-run-server
+      if: inputs.install-packages == 'true' && inputs.build-and-run-server != 'true'
+      shell: bash
+      run: |
+        npm i --prefix=app/server
+        npm i --prefix=app/static
```

Comment:
I guess some jobs only require one of server or static packages (e.g. unit tests) so could make that a parameter too?

Reply:
done have made it a param in the install npm packages action